common/config/persistence.go

```diff
@@ -90,11 +95,7 @@ func (c *Persistence) validateAdvancedVisibility() error {
 		return fmt.Errorf("persistence config: advanced visibility datastore %q: missing config", c.AdvancedVisibilityStore)
 	}
 
-	if advancedVisibilityDataStore.ElasticSearch == nil {
-		return fmt.Errorf("persistence config: advanced visibility datastore %q: must provide config for elasticsearch store", c.AdvancedVisibilityStore)
-	}
-
-	if err := advancedVisibilityDataStore.ElasticSearch.validate(c.AdvancedVisibilityStore); err != nil {
+	if err := advancedVisibilityDataStore.Elasticsearch.Validate(c.AdvancedVisibilityStore); err != nil {
```

Comment:
can just do:
return advancedVisibilityDataStore.Elasticsearch.Validate(c.AdvancedVisibilityStore)

Reply:
I disagree, but it is ok. :)